### ISSUE TITLE
Uri-Query Options not calculated correctly

### ISSUE THREAD

**Issue by holomekc** [AlCalzone/node-coap-client#240]
When I specified an url with the following format:
coap://host:port/a/b/c

The uri-Path options are correct and also Content-Format. But then there is an empty Uri-Query option at the end:

![image](https://user-images.githubusercontent.com/30546982/80276989-55510b80-86ec-11ea-806c-2522abf07246.png)

There are no whitespaces etc. at the end. The last query.split('&'); seems to create: [''] which is then added.

**Comment by AlCalzone** [comment]
Do you by any chance have some code so I can reproduce this easier?

**Comment by holomekc** [comment]
If you mean on github? Then no. Does this help?

```TypeScript
CoapClient.observe('coap://localhost:5683/a/b/c', 'get', resp => {
  console.log(resp);
}, undefined, {
  confirmable: false
}).then(() => {
  // do nothing
}).catch(reason => {
  // handle error
});
```

**Comment by AlCalzone** [comment]
Ok I can reproduce it on my end. Thanks for the report!

**Comment by AlCalzone** [comment]
Ok I refactored the code that generates the Uri-Query option a while back, but I only did it for requests, and forgot the observe calls. This is fixed in `v1.0.5` (available in a couple of minutes).

**Comment by holomekc** [comment]
Thanks a lot!
